src/status_im/chat/db.cljs

```diff
@@ -0,0 +1,341 @@
+        ;; Have we seen an outgoing message already?
+        last-outgoing?           (and (not last-outgoing-seen)
+                                      outgoing)
+        datemark?                (= :datemark (:type message))
+        ;; If this is a datemark or this is the last-message of a group,
+        ;; then the previous message was the first
+        previous-first-in-group? (or datemark?
+                                     last-in-group?)
+        new-message              (assoc message
+                                        :same-direction? same-direction?
+                                        :last-in-group? last-in-group?
+                                        :last-outgoing? last-outgoing?)]
+    {:stream             (cond-> stream
+                           previous-first-in-group?
+                           ;; update previuous message if necessary
+                           set-previous-message-first-in-group
+
+                           :always
+                           (conj new-message))
+     ;; mark the last message sent by the user
+     :last-outgoing-seen (or last-outgoing-seen last-outgoing?)}))
+
+(defn messages-stream
+  "Enhances the messages in message sequence interspersed with datemarks
+  with derived stream context information, like:
+  q  `:first-in-group?`, `last-in-group?`, `:same-direction?`, `:last?` and `:last-outgoing?` flags."
+  [ordered-messages]
+  (when (seq ordered-messages)
+    (let [initial-message (first ordered-messages)
+          message-with-metadata (assoc initial-message
+                                       :last-in-group? true
+                                       :last? true
+                                       :display-photo? (display-photo? initial-message)
+                                       :last-outgoing? (:outgoing initial-message))]
+      (->> (rest ordered-messages)
+           (reduce add-positional-metadata
+                   {:stream             [message-with-metadata]
+                    :last-outgoing-seen (:last-outgoing? message-with-metadata)})
+           :stream))))
+
+(defn get-current-chat-messages-stream
+  [messages messages-statuses referenced-messages
+   current-chat current-network contacts blocked-contacts account]
+  (-> messages
+      (messages-with-metadata current-chat messages-statuses referenced-messages current-network contacts account)
+      (filter-messages-from-blocked-contacts blocked-contacts)
+      intersperse-datemarks
+      messages-stream))
+
+(defn chat-name [{:keys [group-chat
+                         chat-id
+                         public?
+                         name]}
+                 {contact-name :name}]
+  (cond
+    public?    (str "#" name)
+    group-chat name
+    :else      (or contact-name
+                   (gfycat/generate-gfy chat-id))))
+
+(defn- enrich-active-chat
+  [{:keys [chat-id messages unviewed-messages group-chat
```

Comment:
`unviewed-messages` is never used, there are also other unused symbols in the ns, please check them